components/ui/scroll-throttled.tsx

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from "react";
+import { throttle } from "@martinstark/throttle-ts";
+// import { debounce } from "ts-debounce";
+
+function useDocumentScrollThrottled(callback: {
+  (callbackData: { previousScrollTop: Number; currentScrollTop: Number }): void;
+  (arg0: { previousScrollTop: number; currentScrollTop: number }): void;
+}) {
+  const [, setScrollPosition] = useState(0);
+  let previousScrollTop = 0;
+
+  function handleDocumentScroll() {
+    const { scrollTop: currentScrollTop } = document.documentElement;
+
+    setScrollPosition((previousPosition) => {
+      previousScrollTop = previousPosition;
+      return currentScrollTop;
+    });
+
+    callback({ previousScrollTop, currentScrollTop });
+  }
+
+  const [handleDocumentScrollThrottled] = throttle(handleDocumentScroll, 150);
+
+  useEffect(() => {
+    window.addEventListener("scroll", handleDocumentScrollThrottled);
+
+    return () =>
+      window.removeEventListener("scroll", handleDocumentScrollThrottled);
+  }, []);
+}
```

Comment:
If your new tests for the `global-header.tsx` is comprehensive enough, all these lines should have been covered.

Reply:
@zainfathoni I've deleted this file and merged the function to `global-header.tsx` please check it again mas